google/cloud/pubsub/internal/tracing_batch_sink_test.cc

```diff
@@ -216,6 +217,28 @@ TEST(TracingBatchSink, PublishSpanHasAttributes) {
                                  TestTopic().FullName())))));
 }
 
+TEST(TracingBatchSink, PublishSpanParentIsRoot) {
+  auto span_catcher = InstallSpanCatcher();
+  auto message_span = MakeSpan("test span");
+  auto mock = std::make_unique<pubsub_testing::MockBatchSink>();
+  EXPECT_CALL(*mock, AddMessage(_));
+  EXPECT_CALL(*mock, AsyncPublish)
+      .WillOnce([](google::pubsub::v1::PublishRequest const& request) {
+        EXPECT_THAT(request, IsProtoEqual(MakeRequest(1)));
+        return make_ready_future(make_status_or(MakeResponse(request)));
+      });
+  auto batch_sink = MakeTestBatchSink(std::move(mock));
+  auto initial_spans = {message_span};
+  AddMessages(initial_spans, batch_sink);
+
+  auto response = batch_sink->AsyncPublish(MakeRequest(1)).get();
+  EXPECT_THAT(response, IsOk());
+
+  auto spans = span_catcher->GetSpans();
+  EXPECT_THAT(spans, Contains(AllOf(SpanNamed("test-topic publish"),
+                                    SpanParentIsRoot())));
```

Comment:
... this test isn't actually doing anything. We would need there to be an active span when we create the `publish` span to verify it is made as the root span. But if we do that `cmake-oldest-deps` will start failing.

This should at least be addressed in a comment. I am not sure what I would do. I might skip this expectation for older versions of opentelemetry-cpp.

Reply:
Done. 

It might work for oldest-deps, but the test will be flakey. Just like it is in real life :)